### PR TITLE
improve generic polynomial coverage

### PR DESCRIPTION
**Pull request by GiacomoPope**

### CODE REVIEW

File: tests/test_polynomial_generic.py [hunk 1]
```diff
@@ -49,6 +92,9 @@ def test_add_polynomials(self):
             self.assertEqual(f1 + f2, f2 + f1)
             self.assertEqual(f1 + (f2 + f3), (f1 + f2) + f3)
 
+            f1 += zero
+            self.assertEqual(f1, f1)
```
Comment (tomato42) [inline tests/test_polynomial_generic.py:96]: shouldn't those be different ones? (not `f1` repeated)

File: tests/test_polynomial_generic.py [hunk 2]
```diff
@@ -58,9 +104,14 @@ def test_sub_polynomials(self):
 
             self.assertEqual(f1 - zero, f1)
             self.assertEqual(f3 - f3, zero)
+            self.assertEqual(f3 - 0, f3)
+            self.assertEqual(0 - f3, -f3)
             self.assertEqual(f1 - f2, -(f2 - f1))
             self.assertEqual(f1 - (f2 - f3), (f1 - f2) + f3)
 
+            f1 -= zero
+            self.assertEqual(f1, f1)
```
Comment (tomato42) [inline tests/test_polynomial_generic.py:113]: same here

File: tests/test_polynomial_generic.py [hunk 3]
```diff
@@ -73,6 +124,11 @@ def test_mul_polynomials(self):
             self.assertEqual(f1 * one, f1)
             self.assertEqual(f1 * f2, f2 * f1)
             self.assertEqual(f1 * (f2 * f3), (f1 * f2) * f3)
+            self.assertEqual(2 * f1, f1 + f1)
+            self.assertEqual(2 * f1, f1 * 2)
+
+            f1 *= one
+            self.assertEqual(f1, f1)
```
Comment (tomato42) [inline tests/test_polynomial_generic.py:131]: and here...

File: tests/test_polynomial_generic.py [hunk 4]
```diff
@@ -38,6 +55,32 @@ def test_reduce_coefficients(self):
             f = self.R(coeffs).reduce_coefficients()
             self.assertTrue(all([c < self.R.q for c in f.coeffs]))
 
+    def test_equality(self):
+        for _ in range(100):
+            f1 = self.R.random_element()
+            f2 = -f1
+            self.assertEqual(f1, f1)
```
Comment (tomato42) [inline tests/test_polynomial_generic.py:62]: looks like this one was missed...

Comment (GiacomoPope) [reply tests/test_polynomial_generic.py:62]: This one looks like it's doing what I intended

**Comment by GiacomoPope** [comment]
> Target branch is out-of-sync. Rebase this PR to fix. See Details for more.

This in the CLI doesnt seem that useful. Is there a way to work around this @tomato42 ?

File: tests/test_polynomial_generic.py (hunk 4 above)
Comment (tomato42) [reply tests/test_polynomial_generic.py:62]: ah, ok

**Comment by tomato42** [comment]
when do you see this error?

make sure that your local main and the github main match, double check that the `improve_generic_polynomial_coverage` are the same (I like to use `gitk --all` for that) and if that still doesn't make it clear: do what it suggests: rebase the branch on top of main:
```
git checkout improve_generic_polynomial_coverage
git rebase main
```

**Pull request merged**